app/models/task.py

```diff
@@ -1,5 +1,45 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from datetime import datetime
+from typing import Optional
+from sqlalchemy import ForeignKey
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped["Goal"] = relationship("Goal", back_populates="tasks")
+    
+
+
+    def to_dict(self):
+        task_as_dict = {
+            "id": self.id, 
+            "title": self.title, 
+            "description": self.description, 
+            "is_complete": check_for_completion(self)
```

Comment:
If `check_for_completion` is an instance method, call it as 
```py
            "is_complete": self.check_for_completion()
```

Python will shuffle things around internally.